supybot_fedora/plugin.py

```diff
@@ -701,37 +802,69 @@ def sponsors(self, irc, msg, args, name):
 
         Return the sponsors list for the selected group"""
 
-        try:
-            group = self.fasclient.group_members(name)
-            sponsors = ""
-            for person in group:
-                if person["role_type"] == "sponsor":
-                    sponsors += person["username"] + " "
-                elif person["role_type"] == "administrator":
-                    sponsors += "@" + person["username"] + " "
-            irc.reply("Sponsors for %s: %s" % (name, sponsors))
-        except AppError:
-            irc.reply("There is no group %s." % name)
+        if self.registryValue("use_fasjson"):
+            try:
+                sponsors = self.fasjsonclient.list_group_sponsors(groupname=name).result
+            except fasjson_client.errors.APIError as e:
+                if e.code == 404:
+                    irc.reply(f"Sorry, but group '{name}' does not exist")
+                    return
+                else:
+                    irc.reply("Something blew up, please try again")
+                    self.log.error(e)
+                    return
+
+            irc.reply(
+                f"Sponsors for {name}: {', '.join(s['username'] for s in sponsors)}"
+            )
+        else:
+            try:
+                group = self.fasclient.group_members(name)
+                sponsors = ""
+                for person in group:
+                    if person["role_type"] == "sponsor":
+                        sponsors += person["username"] + " "
+                    elif person["role_type"] == "administrator":
+                        sponsors += "@" + person["username"] + " "
+                irc.reply("Sponsors for %s: %s" % (name, sponsors))
+            except AppError:
+                irc.reply("There is no group %s." % name)
 
     sponsors = wrap(sponsors, ["text"])
 
     def members(self, irc, msg, args, name):
         """<group short name>
 
         Return a list of members of the specified group"""
-        try:
-            group = self.fasclient.group_members(name)
-            members = ""
-            for person in group:
-                if person["role_type"] == "administrator":
-                    members += "@" + person["username"] + " "
-                elif person["role_type"] == "sponsor":
-                    members += "+" + person["username"] + " "
+        if self.registryValue("use_fasjson"):
+            try:
+                sponsors = self.fasjsonclient.list_group_members(groupname=name).result
```

Comment:
This variable should be named `members`.

Reply:
done